minkasi/maps/skymap.py

```diff
@@ -179,8 +179,21 @@ def __init__(
             ymax = 2*np.ceil(ny / multiple)
             ydiff = self.lims[3] - self.lims[2]
             self.lims[3] = self.lims[2] + ydiff * ymax/ny
-
+            
             nx, ny = self.get_npix(pad) #This may be applying pad a bunch of times
+            
+        if square:
+            while nx != ny: #TODO: Fix this bug where it doesn't square right sometimes. This is a one time inefficency and it doesn't take much time anyway but could conceivably run forever.
```

Comment:
Maybe add something to avoid getting stuck in an infinite loop in case things fail? A `max_iters` at which point it either error out our does some sensible fallback like truncating one side axis to make the map square?

Reply:
Yeah definitely.